### ISSUE TITLE
Extend the Compendium Enricher's functionality to be called as a global function

### ISSUE THREAD

**Issue by dylanpiera** [dylanpiera/Foundry-Pokemon-Tabletop-United-System#608]
As is right now the filtering of the tab is called as an anonymous onClick event listener.

Move out all the filterdata code into a simple to use global API so we can configure in-sheet buttons to open the compendium browser with appropriate settings.

F.e., in a Trainer's feats tab it'd have a button that opens the browser and would default to having already filtered exactly based on what their skills are.

Step 1: Create the API for this
Step 2: Add simple button to sheets (no filter or if the filter takes less than 5 min to create it can be included)
Step 3: Add custom filtering data based on the actor of that sheet (nice-to-have, but should be released in a seperate update, likely requires some changes from f.e. your keywords patch)

**Comment by Muhsigbokz** [comment]
1 and 2 can easily be done

regarding 3...
I assume with "filtered exactly based on what their skills are" you mean a Feature-button like PF2e for open feats, right?
![image](https://github.com/dylanpiera/Foundry-Pokemon-Tabletop-United-System/assets/101654076/1196610b-276f-4a00-b4de-bc3b4c5844b7)
Do you envision a button that opens the Edges/Features browser with exactly only the Edges/Features for which the Actor fulfills all prerequisites?

**Comment by dylanpiera** [comment]
Yup. So the final version of this would be that you can filter something by skill.

This is very much a nice to have because we don't currently have an easy implementation of this, and I genuinly don't think it's worth much effort, but if you think it can be done in under an hour go ahead.

Otherwise 100% prioritize the rules journal entry and after that likely some 2e.

**Comment by Muhsigbokz** [comment]
That would be 80% on the BrowserTab, as there is no filtering by skill yet, and I guess given the data basis in the prerequisites field it would require some careful work to capture any way the prereqs have been worded so far.
